src/transformState.js

```diff
@@ -5,7 +5,42 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        const { extraData } = action;
+
+        for (const key in extraData) {
+          state[key] = extraData[key];
+        }
+        break;
+      }
+
+      case 'removeProperties': {
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
```

Comment:
Checking if the property exists before deleting it is redundant. The delete operator does not throw an error if the property does not exist.